### PR TITLE
Fix pivot tables don't show stand-alone values on the upper level grouping

### LINKED ISSUE #15211: Pivot table don't show stand-alone values on the upper level grouping
**Describe the bug**
When using a pivot table, groups with one row are not presented in the total sum on the upper level.

image
image
E.g. here we've lost data for Nov 10, 2016

**Expected behaviour**
All data should be shown on the upper level.

**Information about your Metabase Installation:**
- Metabase version: 0.38.1
- Metabase internal database: H2 

**Severity**
We can't currently use pivot tables because of that. 

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by kulyk**
When using a pivot table, groups with one row are not presented in the total sum on the upper level. Fixes #15211

## To Verify

1. Ask a question → Custom question
2. Sample dataset → Orders
    * Add filter: Orders → "Created at" between Nov 9, 2016, and Nov 11, 2016
    * Add filter: Orders → "Product ID" is NOT `146`
    * Summarize by Sum of Orders → Discount
    * Summarize by Count of rows
    * Break down by Orders → Created At → Day
    * Break down by Orders → Product ID
4. Click "Visualize"
5. Click "Visualization" in the bottom left of the screen, select "Pivot table"
6. Click "Settings" (to the left from "Visualization")
7. Move "Product ID" to "Fields to use for the table rows" section
8. Collapse all subtotal rows by clicking a plus button on the "Created at" column 

### Useful screenshots

<details>
<summary>How should the custom question editor look like</summary>

image
</details>

<details>
<summary>How should the pivot table's visualization settings look like</summary>

image
</details>

<details>
<summary>How to collapse pivot table's total values</summary>

Screenshot 2021-06-14 at 14 00 59
</details>

## Demo

Notice the _before_ state doesn't show a row for "November 10"

### Before

before

### After

after

### Details

It's important not to break subtotals for different levels of pivot tables, so I recorded a video with more examples (with voiceover)

https://user-images.githubusercontent.com/17258145/121924566-b1d61680-cd44-11eb-9cf3-6e1f61d74c73.mov

### CODE REVIEW

**Comment by kulyk** [comment]
I don't have a lot of context around how pivot tables should work, please help me find out if something is wrong.

Example: let's say we have a pivot table and each group has only one row. _Before_ the fix, the pivot table wouldn't show any "total" rows. But _after_ the fix it does and maybe the UI is a bit too massive now. Do we maybe expect it to be this way? Or should we handle this case differently?

Before             |  After
:-------------------------:|:-------------------------:
before-fix  |  after-fix

<details>
<summary>Question config</summary>

question

</details>


<details>
<summary>Pivot table visualization settings</summary>

visualization-settings

</details>

File: frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js [hunk 1]
```diff
@@ -810,6 +810,50 @@ describe("scenarios > visualizations > pivot tables", () => {
         .click();
     }
   });
+
+  it("should not show subtotals for flat tables", () => {
```
Comment (kulyk) [inline frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js:814]: This test ensures subtotals are not displayed if the top-level rows are flat (containing only one child). That's not a new behavior, it exists on `master` already.

Correct             |  Broken
:-------------------------:|:-------------------------:
before-fix  |  after-fix

**Review by flamber — APPROVED** [review]
Nice catch with the flat tables. Code looks simple enough, but probably want other eyes on it.

**Review by alxnddr — APPROVED** [review]
👍

**Pull request merged**